lib/idp/saml/session.go

```diff
@@ -0,0 +1,52 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package saml
+
+import (
+	"net/http"
+)
+
+const (
+	// SAMLSessionCookieName is the name of the SAML IdP session cookie.
+	// This cookie is set by SAML IdP after a successful SAML authentication.
+	SAMLSessionCookieName = "__Host-saml_session"
+)
+
+// SetCookie set's SAML session cookie named __Host-saml_session.
+func SetCookie(w http.ResponseWriter, sessionID string, maxAge int) {
```

Comment:
```suggestion
func SetCookie(w http.ResponseWriter, sessionID string, maxAge time.Duration) {
```

This way there's no question about what the time unit is.

Reply:
This isn't as intuitive as it seems because `maxAge` in the http Cookie is annoying fixed to seconds. No reason we can't use duration, but that's the reason for this choice here.

Reply:
Let's do `maxAgeSeconds int`, then, so it's clear what the unit is.